pkg/cmd/connector/update/update.go

```diff
@@ -2,118 +2,177 @@ package update
+	connectorerror "github.com/redhat-developer/app-services-sdk-go/connectormgmt/apiv1/error"
+	"gopkg.in/AlecAivazis/survey.v1"
 
-	"github.com/redhat-developer/app-services-cli/pkg/shared/connection"
 	"github.com/redhat-developer/app-services-cli/pkg/shared/factory"
-	"github.com/spf13/cobra"
+	connectormgmtclient "github.com/redhat-developer/app-services-sdk-go/connectormgmt/apiv1/client"
 )
 
 type options struct {
-	file string
-
+	name         string
 	outputFormat string
 	f            *factory.Factory
-	id           string
 }
 
-// NewUpdateCommand creates a new command to update a connector
 func NewUpdateCommand(f *factory.Factory) *cobra.Command {
-
 	opts := &options{
 		f: f,
 	}
 
 	cmd := &cobra.Command{
 		Use:     "update",
-		Short:   f.Localizer.MustLocalize("connector.update.cmd.shortDescription"),
-		Long:    f.Localizer.MustLocalize("connector.update.cmd.longDescription"),
-		Example: f.Localizer.MustLocalize("connector.update.cmd.example"),
-		Hidden:  true,
+		Short:   f.Localizer.MustLocalize("connector.create.cmd.shortDescription"),
+		Long:    f.Localizer.MustLocalize("connector.create.cmd.longDescription"),
+		Example: f.Localizer.MustLocalize("connector.create.cmd.example"),
 		Args:    cobra.NoArgs,
 		RunE: func(cmd *cobra.Command, args []string) error {
-
 			validOutputFormats := flagutil.ValidOutputFormats
 			if opts.outputFormat != "" && !flagutil.IsValidInput(opts.outputFormat, validOutputFormats...) {
 				return flagutil.InvalidValueError("output", opts.outputFormat, validOutputFormats...)
 			}
 
-			return runUpdateCommand(opts)
+			return runCreate(opts)
 		},
 	}
-
-	flags := connectorcmdutil.NewFlagSet(cmd, f)
-	flags.StringVar(&opts.file, "file", "", f.Localizer.MustLocalize("connector.file.flag.description"))
-	flags.AddConnectorID(&opts.id)
+	flags := flagutil.NewFlagSet(cmd, f.Localizer)
+	flags.StringVar(&opts.name, "name", "", f.Localizer.MustLocalize("connector.flag.name.description"))
 	flags.AddOutput(&opts.outputFormat)
 
 	return cmd
-
 }
 
-func runUpdateCommand(opts *options) error {
-	f := opts.f
+func runCreate(opts *options) error {
 
-	var conn connection.Connection
-	conn, err := f.Connection()
+	conn, err := opts.f.Connection()
 	if err != nil {
 		return err
 	}
 
-	var specifiedFile *os.File
-	if opts.file == "" {
-		file, err1 := util.CreateFileFromStdin()
-		if err1 != nil {
-			return err
-		}
-		specifiedFile = file
-	} else {
-		if util.IsURL(opts.file) {
-			specifiedFile, err = util.GetContentFromFileURL(f.Context, opts.file)
-		} else {
-			specifiedFile, err = os.Open(opts.file)
-		}
-		if err != nil {
-			return err
-		}
-	}
-	defer specifiedFile.Close()
+	api := conn.API()
 
-	byteValue, err := ioutil.ReadAll(specifiedFile)
-	if err != nil {
-		return err
-	}
-	var connector map[string]interface{}
-	err = json.Unmarshal(byteValue, &connector)
+	var connector *connectormgmtclient.Connector
+
+	connector, err = connectorutil.InteractiveSelect(conn, opts.f)
```

Comment:
Interactive mode without option to use flag. 
Interactive mode is not handled by checking if terminal session enables einteractive mode